xayn-ai/src/ltr/list_net/ndutils/mod.rs

```diff
@@ -52,6 +67,45 @@ pub fn kl_divergence(good_dist: ArrayView1<f32>, eval_dist: ArrayView1<f32>) ->
     )
 }
 
+/// He-Uniform Initializer
+///
+/// Weights for layer `j` are sampled from following normal distribution:
+///
+/// ```ascii
+/// W_j ~ N(μ=0, σ²=2/in_j)
+/// ```
+///
+/// Where `n_j` is the number of input units of this layer.
+/// This means for us `n_j` is the the number of rows of `W_j`.
+///
+/// Furthermore as we want to avoid exceedingly large values
+/// we truncate the normal distribution at 2σ.
+///
+/// Source:
+///
+/// - Website: https://www.cv-foundation.org/openaccess/content_iccv_2015/html/He_Delving_Deep_into_ICCV_2015_paper.html
+/// - Pdf: https://www.cv-foundation.org/openaccess/content_iccv_2015/papers/He_Delving_Deep_into_ICCV_2015_paper.pdf
+pub fn he_normal_weights_init<R>(rng: &mut R, nr_rows: Ix, nr_columns: Ix) -> Array2<f32>
```

Comment:
i'd say we shouldn't use `Ix` as the type (even though it works because it's just a `usize` alias), because it has a very specific meaning as the array dimensionality in `ndarray`. the arrays in this case always have dimension 2 and we actually want to express the shape/size of each dimension.
```suggestion
pub fn he_normal_weights_init<R>(rng: &mut R, nr_rows: usize, nr_columns: usize) -> Array2<f32>
```

Reply:
Ix stands for "array index type" (says the `ndarray` documentation) it's the type used for indexing and describing the length of dimensions. `Ix2` e.g. stands for "two array index type values" (wrapped into a `Dim`).

Similar the passed in `nr_rows`, `nr_columns` are used as `(nr_rows, nr_columns)` to create a new array and the method in `ndarray` using that tuple is based onto `impl IntoDimension for (Ix, Ix)` similar if you index into a array it's based on e.g. `&[Ix]`, `(Ix,Ix)`, etc.

But then the returned shape is `&[usize]` and slicing also doesn't use `Ix` :thinking: .

So while I see where you are coming from (Ix being 1,2,3 dimensions) I would also argue that:

- The usage here is in the boundaries of what it's supposed to be used for, especially given that it's mainly used
  to create a `(Ix,Ix)` tuple to create the shape.
- `ndarray` isn't always fully consistent about the usage of `Ix` and `usize`.

Either way I changed it to use `dim: impl IntoDimension<Dim=Ix2>` instead, so it now mirrors all the
`Array` constructors  (like `from_...` or `zeros`) but limited to `Ix2`.